runtime/lua/vim/lsp/buf.lua

```diff
@@ -76,8 +76,76 @@ end
 ---     - on_list: (function) |lsp-on-list-handler| replacing the default handler.
 ---       Called for any non-empty result.
 function M.definition(options)
+  options = options or {}
   local params = util.make_position_params()
-  request_with_options(ms.textDocument_definition, params, options)
+  local bufnr = api.nvim_get_current_buf()
+
+  ---@param results table<integer, {error: lsp.ResponseError, result: lsp.Location|lsp.Location[]|lsp.LocationLink[]|nil}>
+  local function handler(results)
+    local client_id_to_items = {}
```

Comment:
With https://github.com/neovim/neovim/pull/26694 merged you could add a type annotation for this:

```suggestion
  local function handler(results)
    ---@type table<integer, vim.lsp.util.LocationItem[]>
    local client_id_to_items = {}
```